src/routes/inbox.rs

```diff
@@ -32,30 +35,28 @@ pub async fn post(
     Extension(state): Extension<Arc<State>>,
     Json(req): Json<InboxRequest>,
 ) -> Result<extractors::Activity<Value>> {
-    use ActivityType::*;
-
     let actor = state.client.get_actor(&req.actor).await?;
 
     validate_signature(&actor, "post", uri.path(), &headers)?;
-    validate_request(&actor, req.ty, &state).await?;
+    validate_request(&actor, &req.ty, &state).await?;
 
-    match req.ty {
-        Announce | Create => handle_relay(&actor, req.activity, &host, state).await?,
-        Delete | Update => handle_forward(&actor, req.activity, state).await?,
-        Follow => handle_follow(&actor, req.activity, &host, state).await?,
-        Undo => handle_undo(&actor, req.activity, state).await?,
+    match &*req.ty {
+        "Announce" | "Create" => handle_relay(&actor, req.activity, &host, state).await?,
```

Comment:
I'm not as up to speed on the spec as you are I suspect: are we guaranteed on the casing here? Or should we be doing something like lowercasing before comparing?

Reply:
i don't know either. let's keep this as it matches the examples from the spec (see rustypub) until we find out.